File: backends/fastapi/app/routers/health.py
"""
健康检查路由
GET /api/v1/health
"""
from datetime import datetime, timezone

from fastapi import APIRouter

from app.schemas import ApiResponse

router = APIRouter(prefix="/api/v1", tags=["health"])


@router.get("/health")
def health() -> ApiResponse[dict]:
    return ApiResponse.ok({
        "status": "UP",
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "techStack": {
            "framework": "FastAPI >=0.115",
            "language": "Python 3.12",
            "database": "SQLite"
        }
    })

File: backends/elysia/src/routes/health.ts
/**
 * 健康检查路由
 * GET /api/v1/health
 */
import Elysia from "elysia";
import { HealthData, ApiResponse } from "../schemas";

export const healthRoutes = new Elysia({ prefix: "/api/v1" }).get(
  "/health",
  () => {
    return {
      success: true,
      data: {
        status: "UP",
        timestamp: new Date().toISOString(),
        techStack: {
          framework: "Elysia 1.x",
          language: "TypeScript/Bun",
          database: "SQLite",
        },
      },
      message: null,
      errorCode: null,
    };
  },
  {
    response: ApiResponse(HealthData),
    detail: {
      tags: ["System"],
      summary: "健康检查",
    },
  }
);
